database/scrapers/organizations.ts

```diff
@@ -0,0 +1,211 @@
+import { Readable } from "node:stream";
+
+import { BaseScraperModule } from "#commands/db_scrape";
+import { OrganizationSource } from "#enums/organization_source";
+import { OrganizationType } from "#enums/organization_type";
+import StudentOrganization from "#models/student_organization";
+import StudentOrganizationLink from "#models/student_organization_link";
+import StudentOrganizationTag from "#models/student_organization_tag";
+import FilesService from "#services/files_service";
+
+interface DirectusResponse<T> {
+  data: T[];
+}
+
+interface DirectusTag {
+  id: number;
+  name: string;
+}
+
+interface DirectusLink {
+  id: number;
+  name: string;
+  link: string;
+  scientific_circle_id: number | null;
+}
+
+interface DirectusOrganization {
+  id: number;
+  date_created: string;
+  date_updated: string;
+  name: string;
+  logo: string | null;
+  cover: string | null;
+  description: string | null;
+  type: string;
+  source: string;
+  shortDescription: string | null;
+  department: number | null;
+  useCoverAsPreviewPhoto: boolean;
+  isStrategic: boolean;
+  tags: number[];
+  links: number[];
+}
+
+interface FileMetaResponse {
+  data: { filename_disk: string };
+}
+
+interface DirectusTagPivot {
+  id: number;
+  Scientific_Circles_id: number | null;
+  Tags_id: number | null;
+}
+
+export default class OrganizationsScraper extends BaseScraperModule {
+  static name = "Organizations";
+  static description =
+    "populates all required tables for organizations (tags, links,...)";
+  static taskTitle = "Scraping organizations";
+  private filesService = new FilesService();
+  private readonly urls = {
+    orgs: "https://admin.topwr.solvro.pl/items/Scientific_Circles?limit=-1",
+    tags: "https://admin.topwr.solvro.pl/items/Tags",
+    pivot_tags:
+      "https://admin.topwr.solvro.pl/items/Scientific_Circles_Tags?limit=-1",
+    links:
+      "https://admin.topwr.solvro.pl/items/Scientific_Circles_Links?limit=-1",
+  };
+
+  public async run() {
+    const [orgs, tags, links, tagsPivot] = (await Promise.all([
+      this.fetchJSON(this.urls.orgs, "organizations"),
+      this.fetchJSON(this.urls.tags, "tags"),
+      this.fetchJSON(this.urls.links, "links"),
+      this.fetchJSON(this.urls.pivot_tags, "pivot tags"),
+    ])) as [
+      DirectusResponse<DirectusOrganization>,
+      DirectusResponse<DirectusTag>,
+      DirectusResponse<DirectusLink>,
+      DirectusResponse<DirectusTagPivot>,
+    ];
+    this.logger.info("Fetching...");
```

Comment:
consider using `task.update()` instead of `this.logger.info()` for status updates